core/primitives-core/src/account.rs

```diff
@@ -67,13 +68,49 @@ impl AccountV1 {
         AccountV2 {
             amount: self.amount,
             locked: self.locked,
-            code_hash: self.code_hash,
             storage_usage: self.storage_usage,
+            account_contract: convert_code_hash(self.code_hash),
         }
     }
 }
 
-// TODO(global-contract): add new field
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    serde::Serialize,
+    serde::Deserialize,
+    PartialEq,
+    Eq,
+    Debug,
+    Clone,
+    ProtocolSchema,
+)]
+pub enum AccountContractType {
```

Comment:
just `AccountContract` and no need to repeat `Contract` in each variant: `LocalContract` -> `Local`